From f50df393bf8bc0bd9cf31adc2695a55893b401da Mon Sep 17 00:00:00 2001
From: alfsamsung <alfsamsung@hotmail.com>
Date: Mon, 1 Dec 2014 17:29:10 +0100
Subject: [PATCH] Keydisabler: fix button light restore logic

Do not assign default button brightness if backup key is not found.
Instead do nothing. This will prevent brightness being reset to default
on boot when navbar is disabled.

If backup key is not found when actually toggling navbar then we've got
bigger problems.

Change-Id: I649bf1480a42f2cbe94d0a47a0c6055c934b4f56
---
 src/com/android/settings/mahdi/HardwareKeys.java | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/src/com/android/settings/mahdi/HardwareKeys.java b/src/com/android/settings/mahdi/HardwareKeys.java
index 5970070..bb54eb1 100644
--- a/src/com/android/settings/mahdi/HardwareKeys.java
+++ b/src/com/android/settings/mahdi/HardwareKeys.java
@@ -518,10 +518,12 @@ public class HardwareKeys extends SettingsPreferenceFragment implements
             Settings.System.putInt(context.getContentResolver(),
                     Settings.System.BUTTON_BRIGHTNESS, 0);
         } else {
-            Settings.System.putInt(context.getContentResolver(),
-                    Settings.System.BUTTON_BRIGHTNESS,
-                    prefs.getInt("pre_navbar_button_backlight", defaultBrightness));
-            editor.remove("pre_navbar_button_backlight");
+            int oldBright = prefs.getInt("pre_navbar_button_backlight", -1);
+            if (oldBright != -1) {
+                Settings.System.putInt(context.getContentResolver(),
+                        Settings.System.BUTTON_BRIGHTNESS, oldBright);
+                editor.remove("pre_navbar_button_backlight");
+            }
         }
         editor.commit();
     }
-- 
1.9.1

